chain/network/src/recorder.rs

```diff
@@ -0,0 +1,218 @@
+use crate::types::PeerMessage;
+use actix::Message;
+use near_primitives::{hash::CryptoHash, network::PeerId};
+use serde::ser::SerializeMap;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use tracing::info;
+
+#[derive(Clone, Copy)]
+pub enum Status {
+    Sent,
+    Received,
+}
+#[derive(Default, Serialize, Deserialize, Debug, Clone)]
+struct CountSize {
+    count: usize,
+    bytes: usize,
+}
+
+impl CountSize {
+    fn update(&mut self, bytes: usize) {
+        self.count += 1;
+        self.bytes += bytes;
+    }
+}
+
+#[derive(Default, Serialize, Deserialize, Debug, Clone)]
+struct SentReceived {
+    sent: CountSize,
+    received: CountSize,
+}
+
+impl SentReceived {
+    fn get(&mut self, status: Status) -> &mut CountSize {
+        match status {
+            Status::Sent => &mut self.sent,
+            Status::Received => &mut self.received,
+        }
+    }
+}
+
+// TODO(MarX): Use more compact debug. Right now it will print all known hashes.
+#[derive(Default, Debug, Deserialize, Clone)]
+struct HashAggregator {
+    total: usize,
+    all: HashSet<CryptoHash>,
+}
+
+impl HashAggregator {
+    fn add(&mut self, hash: CryptoHash) {
+        self.total += 1;
+        self.all.insert(hash);
+    }
+
+    fn different(&self) -> usize {
```

Comment:
this name is confusing.

Reply:
fair enough. It is referring to the total number of different blocks. 

This is the most important metric I'm willing to see in the 100 nodes test btw, since I think the network traffic can be decreased noticeably by reducing the number of repeated received blocks.